modules/onewire/onewire.h

```diff
@@ -0,0 +1,287 @@
+ * @return retval: see defintion of OneWireStatus
+ */
+OneWireStatus OneWire_ResetSearch(OneWire_t* OneWireStruct);
+
+/**
+ * @brief  Starts search, reset states first
+ * @note   When you want to search for ALL devices on one onewire port, you should first use this function.
+\code
+//...Initialization before
+status = OneWire_First(&OneWireStruct);
+while (status) {
+    //Save ROM number from device
+    OneWire_GetFullROM(ROM_Array_Pointer);
+    //Check for new device
+    status = OneWire_Next(&OneWireStruct);
+}
+\endcode
+ * @param  *OneWireStruct: Pointer to an initialized @ref Onewire_t structure
+ * @return retval: see defintion of OneWireStatus
+*/
+OneWireStatus OneWire_First(OneWire_t* OneWireStruct);
+
+/**
+ * @brief  Reads next device
+ * @note   Use @ref OneWire_First to start searching
+ * @param  *OneWireStruct: Pointer to an initialized @ref Onewire_t structure
+ * @return retval: see defintion of OneWireStatus
+ */
+OneWireStatus OneWire_Next(OneWire_t* OneWireStruct);
+
+/**
+ * @brief  Gets ROM number from device from search
+ * @param  *OneWireStruct: Pointer to an initialized @ref Onewire_t structure
+ * @param  *ROM : Pointer to store ROM value
+ * @param  index: Because each device has 8-bytes long ROM address, you have to call this 8 times, to get ROM bytes from 0 to 7
+ * @return retval: see defintion of OneWireStatus
+ */
+OneWireStatus OneWire_GetROM(OneWire_t* OneWireStruct, uint8_t index, uint8_t *ROM);
+
+/**
+ * @brief  Gets all 8 bytes ROM value from device from search
+ * @param  *OneWireStruct: Pointer to an initialized @ref Onewire_t structure
+ * @param  *firstIndex: Pointer to first location for first byte, other bytes are automatically incremented
+ * @return retval: see defintion of OneWireStatus
+ */
+OneWireStatus OneWire_GetFullROM(OneWire_t* OneWireStruct, uint8_t *firstIndex);
+
+/**
+ * @brief  Selects specific slave on bus
+ * @param  *OneWireStruct: Pointer to an initialized @ref Onewire_t structure
+ * @param  *addr: Pointer to first location of 8-bytes long ROM address
+ * @return retval: see defintion of OneWireStatus
+ */
+OneWireStatus OneWire_Select(OneWire_t* OneWireStruct, uint8_t* addr);
+
+/**
+ * @brief  Selects specific slave on bus with pointer address
+ * @param  *OneWireStruct: Pointer to an initialized @ref Onewire_t structure
+ * @param  *ROM: Pointer to first byte of ROM address
+ * @return retval: see defintion of OneWireStatus
+ */
+OneWireStatus OneWire_SelectWithPointer(OneWire_t* OneWireStruct, uint8_t* ROM);
+
+/**
+ * @brief  Calculates 8-bit CRC for 1-wire devices
+ * @param  *addr:  Pointer to 8-bit array of data to calculate CRC
+ * @param  len:    Number of bytes to check
+ * @param  *rslt:  Pointer to store calculated CRC value 
+ * @return retval: see defintion of OneWireStatus
+ */
+OneWireStatus OneWire_CalculateCRC8(uint8_t* addr, uint8_t len, uint8_t* rslt);
+
+/**
+ * @brief  Looks up 8-bit CRC for 1-wire devices, in an effort to save MIPS

```

Comment:
It would be good practice to state what CRC polynomial you are using or state which CRC algorithm is being used (which one could then look up the polynomial, if they wanted).